Introduce labels for injecting Linux capabilities.

clems4ever opened the pull request:
This patch only allows to set network capabilities for the moment.

clems4ever committed:
Introduce labels for injecting Linux capabilities.

This patch only allows to set network capabilities.

clems4ever committed:
Add .travis.yml for building, testing and publishing the artifact.

clems4ever commented:
cc @aboten

kamaradclimber commented on src/test/scala/com/criteo/marathon/CapabilitiesPluginSpec.scala:
```diff
@@ -0,0 +1,209 @@
+            .setValue("./a_command"))
+          taskInfoBuilder.getLabelsBuilder.addLabels(Label.newBuilder()
+            .setKey("MESOS_BOUNDING_CAPABILITIES")
+            .setValue("CUSTOM_CAPABILITY;UNKNOWN_CAPA"))
```
I think it should crash the task (not start it) otherwise typo might not get caught

clems4ever replied:
Agreed.

clems4ever replied:
Fixed, I eventually decided to modify the task to write an error message in stderr and fail instead of throwing an exception in Marathon because it induced a weird behavior in the deployment and did not bring any value in the UI anyway.

clems4ever committed:
Make the command fail when using unknown capabilities.

In order to help users figure out that there is an error in the
configuration of the capabilities, the task is modified to print
an error message in stderr and fail.

Also replace ';' by ',' to list capabilities.

kamaradclimber commented on src/main/scala/com/criteo/marathon/CapabilitiesPlugin.scala:
```diff
@@ -0,0 +1,111 @@
+
+    if (unknownBoundingCapabilities || unknownEffectiveCapabilities) {
+      builder.getCommandBuilder.setValue(
+        "echo 'Unknown capabilities. Please double check your configuration.' >&2 && exit 1")
```
maybe we should display the capabilities we were unable to recognize?

clems4ever replied:
Done
